### PR TITLE
Updated XSockets code for realtime .NET sample

### PR DESCRIPTION
**Pull request by uffebjorklund**

### CODE REVIEW

**Comment by leggetter** [comment]
Thanks for this @uffebjorklund. One thing I need to do is go through the presentation and see what points I'm trying to highlight with the XSockets example. It may have been specific communication patterns and I'm not sure if this changes those.

Will let you know.

**Comment by uffebjorklund** [comment]
Thanks @leggetter - It might change things if you were showing simple messaging with XSockets since I changed it to RPC/RMI. Let me know if you I can help in any way. If you want to show different comm-patterns I can write an example showing simple, pub/sub, rpc/rmi and datasync with XSockets.

File: .vs/config/applicationhost.config [hunk 1]
```diff
@@ -0,0 +1,1027 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (leggetter) [inline .vs/config/applicationhost.config:1]: @uffebjorklund is this file required to be checked in or should everything in `.vs` be ignored?

**Comment by leggetter** [comment]
@uffebjorklund - I think I'm going to leave as-is for DevWeek. I'm going to demonstrate RMI/RPC with XSockets and PubSub. The current example does that:
1. Fetching initial messagings
2. New messages via PubSub

Cheers for the PR though.

**Comment by uffebjorklund** [comment]
Understood! It feels weird to deserialize the data in OnMessage instead of having a method that received the expected type (or have several parameters). It's like you allow anything to be sent in, but then inside you expect it to be of a certain type. But that is of course up to you.

I would do atleast this to avoid the deserialize in the OnMessage method that will receive anything that will not match another method on the controller.

```
public async Task ChatMessage(string user, string text)
{
    var messageModel = this.repo.CreateMessage(user, text);
    await this.InvokeToAll(messageModel, "newmessage");            
}
```

**Comment by leggetter** [comment]
> It feels weird to deserialize the data in OnMessage instead of having a method that received the expected type (or have several parameters)

Completely agree. I'd much prefer the code to be:

``` cs
       public async Task ChatMessage(string username, string text)
        {
            var messageModel = this.repo.CreateMessage(username, text);
            await this.PublishToAll(messageModel, "chat");
        }
```

So a PR changing that would be super-awesome! Although is there a way of getting the "chat" value from a property on something?
